Refactor local runner and add `run` command (#234)
ml6team/fondant: README.md

### 🧱 Deploying Fondant

There are 2 ways of using fondant:

> - Leveraging [kubeflow pipelines](https://www.kubeflow.org/docs/components/pipelines/v1/introduction/) on any kubernetes cluster. All fondant needs is an url pointing to the kubeflow pipeline host and a Object Storage provider (S3, GCS, etc) to store data produced in the pipeline between steps.

Reviewer: ```suggestion
- Leveraging [kubeflow pipelines](https://www.kubeflow.org/docs/components/pipelines/v1/introduction/) on any kubernetes cluster. All that fondant needs is an url pointing to the kubeflow pipeline host and an Object Storage provider (S3, GCS, etc) to store data produced in the pipeline between steps.
```

> - Leveraging [kubeflow pipelines](https://www.kubeflow.org/docs/components/pipelines/v1/introduction/) on any kubernetes cluster. All fondant needs is an url pointing to the kubeflow pipeline host and a Object Storage provider (S3, GCS, etc) to store data produced in the pipeline between steps.

Reviewer 2: ```suggestion
- Leveraging [Kubeflow pipelines](https://www.kubeflow.org/docs/components/pipelines/v1/introduction/) on any Kubernetes cluster. All Fondant needs is a url pointing to the Kubeflow pipeline host and an Object Storage provider (S3, GCS, etc) to store data produced in the pipeline between steps.
```

  We have compiled some references and created some scripts to [get you started](https://fondant.readthedocs.io/en/latest/infrastructure) with setting up the required infrastructure.

> - Or locally by using [docker compose](https://docs.docker.com/compose/). This way is mainly aimed at helping you develop fondant pipelines and components faster by making it easier to run things on a smaller scale.

Reviewer 2: ```suggestion
- Or locally by using [Docker Compose](https://docs.docker.com/compose/). This way is mainly aimed at helping you develop Fondant pipelines and components faster by making it easier to run things on a smaller scale.
```

> The same pipeline can be used in both variant allowing you to quickly develop and iterate using the local docker compose implementation and then using the power of kubeflow pipelines to run a large scale pipeline.

Reviewer: ```suggestion
The same pipeline can be used in both variants, allowing you to quickly develop and iterate using the local docker compose implementation and then using the power of kubeflow pipelines to run a large scale pipeline.
```

> The same pipeline can be used in both variant allowing you to quickly develop and iterate using the local docker compose implementation and then using the power of kubeflow pipelines to run a large scale pipeline.

Reviewer 2: ```suggestion
The same pipeline can be used in both variant allowing you to quickly develop and iterate using the local Docker Compose implementation and then using the power of Kubeflow pipelines to run a large scale pipeline.
```

        """
```

###  Running your pipeline

> Once you have a pipeline you can easily run (and compile it) using the builtin CLI:

Reviewer: ```suggestion
Once you have a pipeline you can easily run (and compile) it by using the built-in CLI:
```

```bash

> fondant run foo/bar.py:pipeline --local

Reviewer: This is for the --local, should we add both?

Author: the kubeflow command is there but raises NotImplementedError's

> fondant run foo/bar.py:pipeline --local

Reviewer 3: ```suggestion
fondant run foo.bar:pipeline --local
```
You should pass the file as a module, right?

> fondant run foo.bar.py:pipeline --local

Reviewer 3: ```suggestion
fondant run foo.bar:pipeline --local
```
The `.py` should also be removed.
